pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/DistinctCountHLLAggregationFunction.java

```diff
@@ -368,6 +370,23 @@ public Long mergeFinalResult(Long finalResult1, Long finalResult2) {
     return finalResult1 + finalResult2;
   }
 
+  @Override
+  public boolean canUseStarTree(AggregationFunctionColumnPair functionColumnPair,
+      Map<String, Object> functionParameters) {
+    if (!super.canUseStarTree(functionColumnPair, functionParameters)) {
+      return false;
+    }
+
+    // Check if log2m matches
+    if (functionParameters.containsKey(Constants.HLL_LOG2M_KEY)) {
```

Comment:
For map value check, we can always save one map lookup by directly calling `get()` and then check if the returned value is null. Same for other places